File: AspCoreSandbox/ClientApp/app/shared/product.ts

/** Tworzymy ścisły Typ dla produktu */
export interface Product {
    id: number;
    category: string;
    size: string;
    price: number;
    title: string;
    description: string;
    dateOfProduceStart: Date;
    dateOfProduceEnd: Date;
    countryOfOrigin: string;
}

File: AspCoreSandbox/Data/Entities/Product.cs
using System;

namespace AspCoreSandbox.Data.Entities
{
    public class Product
  {
    public int Id { get; set; }
    public string Category { get; set; }
    public string Size { get; set; }
    public decimal Price { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DateOfProduceStart { get; set; }
    public DateTime DateOfProduceEnd{ get; set; }
    public string CountryOfOrigin { get; set; }
  }
}
